### PR TITLE
50083_create_activities

### PR DESCRIPTION
**Pull request by mdmahmudurrahman**

### CODE REVIEW

**Comment by shakil-fg** [comment]
ok

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -7,6 +7,8 @@ def index
   end
 
   def show
+    @user_activities = PublicActivity::Activity.order("created_at desc").
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: you can move where to scope in Public activity?

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,6 +1,7 @@
 class ApplicationController < ActionController::Base
   before_action :set_locale
   protect_from_forgery with: :exception
+  include PublicActivity::StoreController
   include CanCan::ControllerAdditions
   rescue_from CanCan::AccessDenied do |exception|
     flash[:error] = I18n.t("access_denied")
```
Comment (shakil-fg) [inline app/controllers/application_controller.rb]: remove ()

use symbol

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -7,6 +7,7 @@ def index
   end
 
   def show
+    @user_activities = User.get_review_activities(current_user)
```
Comment (shakil-fg) [inline app/controllers/users_controller.rb]: remove ()

File: app/models/user.rb [hunk 4]
```diff
@@ -27,4 +27,7 @@ class User < ActiveRecord::Base
 
   mount_uploader :picture, ImageUploader
   validate :picture_size
+
+  scope :get_review_activities, ->(user){ PublicActivity::Activity.order("created_at desc").
+      where("trackable_type = ? AND owner_id = ?","Review", user.id)}
```
Comment (shakil-fg) [inline app/models/user.rb]: remove ()

**Comment by MuhammadTamzid** [comment]
ok

File: app/models/user.rb [hunk 5]
```diff
@@ -27,4 +27,7 @@ class User < ActiveRecord::Base
 
   mount_uploader :picture, ImageUploader
   validate :picture_size
+
+  scope :get_review_activities, ->(user){ PublicActivity::Activity.order("created_at desc").
```
Comment (dieunb) [inline app/models/user.rb]:
```
+  scope :review_activities, ->(user){PublicActivity::Activity.order(created_at: :desc).
 +      where(trackable_type: "Review", owner_id: user.id)}
```

File: app/models/user.rb [hunk 6]
```diff
@@ -27,4 +27,7 @@ class User < ActiveRecord::Base
 
   mount_uploader :picture, ImageUploader
   validate :picture_size
+
+  scope :review_activities, ->(user){ PublicActivity::Activity.order(created_at: :desc).
```
Comment (dieunb) [inline app/models/user.rb]: `{PublicActivity::Activity.order(created_at: :desc).`

File: app/views/users/_user_activity.html.erb [hunk 7]
```diff
@@ -0,0 +1,13 @@
+<% if @user_activities.any? %>
+  <% @user_activities.each do |activity| %>
+    <%= activity.owner.name if activity.owner %>
+      <% if activity.key == "review.create" %>
+        <%= t :created_reviews_for %>
+          <%= link_to activity.trackable.book.title, book_path(activity.trackable.book) if activity.trackable %>
```
Comment (dieunb) [inline app/views/users/_user_activity.html.erb]: indent

File: app/views/users/_user_activity.html.erb [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+      <% else %>
+        <%= t :updated_reviews_for %>
+          <%= activity.trackable.book.title if activity.trackable %>
```
Comment (dieunb) [inline app/views/users/_user_activity.html.erb]: indent

**Comment by shakil-fg** [comment]
ok

**Pull request merged**